src/app/main.rs

```diff
@@ -522,7 +522,10 @@ where
                 .layer(buffer::layer(max_in_flight, DispatchDeadline::extract));
 
             let balancer_stack = svc::builder()
-                .layer(fallback::layer(balancer, orig_dst_router))
+                .layer(
+                    fallback::layer(balancer, orig_dst_router)
+                        .on_error::<control::destination::Unresolvable>(),
+                )
```

Comment:
tioli: consider adding a `fallback_on` combinator to the svc::Builder so this could be...

```
￼                 .layer(balancer)
                 .fallback_on::<control::destination::Unresolvable>(orig_dst_router)
```

perhaps not worth it since it's only used once atm, but seems like a small readability improvement.

Reply:
oh, yeah, that is rather nice. good call!